ray-operator/pkg/features/features.go

```diff
@@ -0,0 +1,48 @@
+package features
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+)
+
+const (
+	// owner: @rueian @kevin85421 @andrewsykim
+	// rep: TODO
+	// alpha: v1.2
+	//
+	// Enables new conditions in RayCluster status
+	RayClusterStatusConditions featuregate.Feature = "RayClusterStatusConditions"
+)
+
+func init() {
+	runtime.Must(utilfeature.DefaultMutableFeatureGate.Add(defaultFeatureGates))
+}
+
+var defaultFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
+	RayClusterStatusConditions: {Default: false, PreRelease: featuregate.Alpha},
```

Comment:
We can turn it on by default for the alpha release if we successfully deliver it before v1.2.0. Unlike a big project like Kubernetes, if we don't turn on a feature, it is hard for us to collect feedback because no user will turn it on.

Reply:
If we plan to turn feature gates on by default in alpha, then we probably don't need feature gates? 

The main purpose of the feature gate is to introduce early versions of features in alpha that we can break / change before going to Beta / GA. Agree that significantly less users will use alpha features if disabled by default, but that's the trade-off of being able to break alpha stage features. If we enable by default, then breaking a feature in alpha phase will break many users

Reply:
How about we turn it off at this moment? We can try to find some users to try the alpha feature. If we get enough users to test this alpha feature, we can turn it off by default.

Reply:
If we don't find enough users, we can turn it on by default to collect user feedback more aggressively.

Reply:
> How about we turn it off at this moment? We can try to find some users to try the alpha feature. If we get enough users to test this alpha feature, we can turn it off by default.

This sounds good to me